tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/AWS/EventBridge/AwsEventBridgeCommon.cs

```diff
@@ -0,0 +1,85 @@
+{
+    internal static class AwsEventBridgeCommon
+    {
+        private const string DatadogAwsEventBridgeServiceName = "aws-eventbridge";
+        private const string EventBridgeRequestOperationName = "eventbridge.request";
+        private const string EventBridgeServiceName = "EventBridge";
+        private const string EventBridgeOperationName = "aws.eventbridge";
+        private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor(typeof(AwsEventBridgeCommon));
+
+        internal const string IntegrationName = nameof(IntegrationId.AwsEventBridge);
+        private const IntegrationId IntegrationId = Configuration.IntegrationId.AwsEventBridge;
+
+        public static Scope? CreateScope(Tracer tracer, string operation, string spanKind, out AwsEventBridgeTags? tags, ISpanContext? parentContext = null)
+        {
+            tags = null;
+
+            if (!tracer.Settings.IsIntegrationEnabled(IntegrationId) || !tracer.Settings.IsIntegrationEnabled(AwsConstants.IntegrationId))
+            {
+                // integration disabled, don't create a scope, skip this trace
+                return null;
+            }
+
+            Scope? scope = null;
+
+            try
+            {
+                tags = tracer.CurrentTraceSettings.Schema.Messaging.CreateAwsEventBridgeTags(spanKind);
+                var serviceName = tracer.CurrentTraceSettings.GetServiceName(tracer, DatadogAwsEventBridgeServiceName);
+                var operationName = GetOperationName(tracer);
+                scope = tracer.StartActiveInternal(operationName, parent: parentContext, tags: tags, serviceName: serviceName);
+                var span = scope.Span;
+
+                span.Type = SpanTypes.Http;
+                span.ResourceName = $"{EventBridgeServiceName}.{operation}";
+
+                tags.Service = EventBridgeServiceName;
+                tags.Operation = operation;
+                tags.SetAnalyticsSampleRate(IntegrationId, tracer.Settings, enabledWithGlobalSetting: false);
+                tracer.TracerManager.Telemetry.IntegrationGeneratedSpan(IntegrationId);
+            }
+            catch (Exception ex)
+            {
+                Log.Error(ex, "Error creating or populating scope.");
+            }
+
+            // always returns the scope, even if it's null because we couldn't create it,
+            // or we couldn't populate it completely (some tags is better than no tags)
+            return scope;
+        }
+
+        // Finds the first entry that has a valid bus name, or null if not found.
+        public static string? GetBusName(IEnumerable? entries)
+        {
+            return entries?
+                  .Cast<object>()
+                  .Select(entry => entry.DuckCast<IPutEventsRequestEntry>())
+                  .Where(duckEntry => duckEntry != null)
+                  .Select(duckEntry => duckEntry!.EventBusName)
+                  .FirstOrDefault(eventBusName => !string.IsNullOrEmpty(eventBusName));
```

Comment:
This is me being a bit nitpicky, but could we change this to be a normal loop? (reason personal preference :) )
Along with a `TryDuckCast` instead of a `DuckCast`? (I think we recommend `DuckCast`, but `TryDuckCast` is safer as it shouldn't throw)

```suggestion
            if (entries is null)
            {
                return null;
            }

            foreach (var entry in entries)
            {
                if (entry.TryDuckCast<IPutEventsRequestEntry>(out var requestEntry)
                    && !string.IsNullOrEmpty(requestEntry.EventBusName))
                {
                    return requestEntry.EventBusName;
                }
            }

            return null;
```

Reply:
Note: I did not test that code

Reply:
Yeah that's definitely more readable. Looking back, the Linq expression here was unnecessary